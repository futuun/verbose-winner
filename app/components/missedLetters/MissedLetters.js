import React, { Component, PropTypes } from 'react'
import styles from './styles.sass'

class MissedLetters extends Component {
  render() {
    const { letters } = this.props

    if (!letters.length) return null

    return (
      <div className={styles.missedLetters}>
        <div className={styles.title}>YOU MISSED:</div>
        <div className={styles.letters}>
          {letters.map((char, i) => (
            <span key={i}>{char}</span>
          ))}
        </div>
      </div>
    )
  }
}

MissedLetters.defaultProps = {
  letters: []
}

MissedLetters.propTypes = {
  letters: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default MissedLetters
