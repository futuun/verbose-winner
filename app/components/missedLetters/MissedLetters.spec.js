import React from 'react'
import renderer from 'react-test-renderer'
import MissedLetters from './MissedLetters'

test('MissedLetters: empty', () => {
  const component = renderer.create(
    <MissedLetters />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('MissedLetters: one letter', () => {
  const component = renderer.create(
    <MissedLetters
      letters={['x']}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('MissedLetters: more than one letter', () => {
  const component = renderer.create(
    <MissedLetters
      letters={['a', 'b', 'c', 'd']}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
