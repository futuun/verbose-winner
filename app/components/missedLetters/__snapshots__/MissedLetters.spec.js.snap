exports[`test MissedLetters: empty 1`] = `null`;

exports[`test MissedLetters: more than one letter 1`] = `
<div
  className="missedLetters">
  <div
    className="title">
    YOU MISSED:
  </div>
  <div
    className="letters">
    <span>
      a
    </span>
    <span>
      b
    </span>
    <span>
      c
    </span>
    <span>
      d
    </span>
  </div>
</div>
`;

exports[`test MissedLetters: one letter 1`] = `
<div
  className="missedLetters">
  <div
    className="title">
    YOU MISSED:
  </div>
  <div
    className="letters">
    <span>
      x
    </span>
  </div>
</div>
`;
